src/simplewallet/simplewallet.cpp

```diff
@@ -2808,6 +2865,56 @@ bool simple_wallet::set_track_uses(const std::vector<std::string> &args/* = std:
   return true;
 }
 
+bool simple_wallet::setup_background_sync(const std::vector<std::string> &args/* = std::vector<std::string>()*/)
+{
+  if (m_wallet->multisig())
+  {
+    fail_msg_writer() << tr("background sync not implemented for multisig wallet");
+    return true;
+  }
+  if (m_wallet->watch_only())
+  {
+    fail_msg_writer() << tr("background sync not implemented for watch only wallet");
+    return true;
+  }
+  if (m_wallet->key_on_device())
+  {
+    fail_msg_writer() << tr("command not supported by HW wallet");
+    return true;
+  }
+
+  const auto pwd_container = get_and_verify_password();
+  if (pwd_container)
+  {
+    tools::wallet2::BackgroundSyncType background_sync_type;
+    if (!parse_background_sync_type(args[1], background_sync_type))
+    {
+      fail_msg_writer() << tr("invalid option");
+      return true;
+    }
```

Comment:
This should be put before `get_and_verify_password()` since it would be frustrating to type your password in and THEN get a parse error.